### PR TITLE
Fix checkbox and select losing value when typing on inputs

### PR DESCRIPTION
**Pull request by shamanime**
This is a resume from https://github.com/phoenixframework/phoenix/pull/5000.

That branch got otudated and the PR was automatically closed becase of a force push when I let GitHub sync the fork.

Fix https://github.com/phoenixframework/phoenix/issues/4998.

This extends https://github.com/phoenixframework/phoenix/pull/5041.

### CODE REVIEW

File: priv/templates/phx.gen.live/core_components.ex [hunk 1]
```diff
@@ -252,9 +252,15 @@ defmodule <%= @web_namespace %>.CoreComponents do
   attr :errors, :list
   attr :rest, :global, include: ~w(autocomplete checked disabled form max maxlength min minlength
                                    multiple pattern placeholder readonly required size step)
+
+  attr :hidden_input, :boolean,
+    default: true,
+    doc: "wether to show or not a hidden input for the unchecked checkbox input"
```
Comment (shamanime) [inline priv/templates/phx.gen.live/core_components.ex:258]: Unchecked checkboxes aren't sent by default so it needs a hidden input with the value of "false" to update the boolean value. This allows having or not having the hidden input.
When working with some forms that don´t use checkboxes for boolean values, the hidden input is optional.

File: lib/mix/tasks/phx.gen.html.ex [hunk 2]
```diff
@@ -184,7 +184,15 @@ defmodule Mix.Tasks.Phx.Gen.Html do
...
 
       {key, :boolean} ->
-        ~s(<.input field={{f, #{inspect(key)}}} type="checkbox" label="#{label(key)}" />)
+        ~s"""
+        <.input
+          field={{f, #{inspect(key)}}}
+          type="checkbox"
+          label="#{label(key)}"
+          value="true"
+          checked={to_string(input_value(f, #{inspect(key)})) == "true"}
```
Comment (shamanime) [inline lib/mix/tasks/phx.gen.html.ex:193]: I tried moving this logic to the component but it ended up being more complicated than having the value and the condition to check here.

Some of the variations I tested was from https://github.com/phoenixframework/phoenix/pull/5000#discussion_r987467759.

**Comment by mveytsman** [comment]
Might be nice to add attrs for `checked_value` and `unchecked_value` in order to mirror [`Phoenix.HTML.Form.checkbox/3`](https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#checkbox/3)

File: installer/templates/phx_web/components/core_components.ex [hunk 3]
```diff
@@ -252,9 +252,15 @@ defmodule <%= @web_namespace %>.CoreComponents do
   attr :errors, :list
   attr :rest, :global, include: ~w(autocomplete checked disabled form max maxlength min minlength
                                    multiple pattern placeholder readonly required size step)
+
+  attr :hidden_input, :boolean,
+    default: true,
+    doc: "wether to show or not a hidden input for the unchecked checkbox input"
```
Comment (lenileiro) [inline installer/templates/phx_web/components/core_components.ex:258]:
```suggestion
    doc: "whether to show or not a hidden input for the unchecked checkbox input"
```

**Comment by shamanime** [comment]
Ugh, I messed up git so hard.

Care to take a look again @mveytsman? Thanks for your suggestion. I implemented it and tested with multiple combinations and looks like it's 👌🏻

**Comment by chrismccord** [comment]
Closed in 9f0b9eca871c9816cd33090177f77b82d152afe6 and I added you as a co-author. Thank you! ❤️❤️❤️🐥🔥

**Pull request closed without merging**